client/App.zig

```diff
@@ -924,10 +986,27 @@ fn updateDisplays(self: *App) !void {
     }
 }
 
-fn updateWindowState(self: *App) !void {
-    self.game.window_state = .{
-        .isFullscreen = self.window.fullscreen,
-    };
+fn getPrimaryDisplay(self: *App) ?renderite.shared.DisplayState {
+    for (self.game.displays.items) |display| {
+        if (display.isPrimary)
+            return display;
+    }
+
+    return null;
+}
+
+fn updatePresentMode(self: *App, present_mode: gpu.PresentMode) !void {
+    if (present_mode == self.window.present_mode)
+        return;
+
+    if (!self.window.window.hasSurface())
+        return;
+
+    if (!self.graphics_data.device.windowSupportsPresentMode(self.window.window, present_mode))
+        return error.PresentModeUnsupported;
```

Comment:
We should default to `vsync` if the present mode is unsupported, since that's the only mode that is guarenteed to be supported.

Reply:
This shouldn't be hit ever, actually.. unless the swapchains can randomly lose support for present modes?

Maybe I should refactor this to be a dedicated setVsync function instead, because what *actually* happens is the caller of this function either sets .vsync or the window's default present mode that was determined on startup. 

Whatever is set in that variable is guaranteed to be supported, so this should actually be `unreachable`.